rmgpy/data/kinetics/family.py

```diff
@@ -1602,6 +1604,24 @@ def apply_recipe(self, reactant_structures, forward=True, unique=True):
                 lowest_labels.append(min(labels))
             product_structures = [s for _, s in sorted(zip(lowest_labels, product_structures))]
 
+        # if applying the family in reverse and the template reactants restrict multiplicity, we need to make sure that
+        # a reverse product matches the multiplicity-constrained template reactant because the forward template products
+        # do not enforce the multiplicity restriction
+        if not forward and reactant_type == 'mol':
+            multiplicity_restricted_templates = []
+            for node in self.forward_template.reactants:
+                if isinstance(node.item, Group):
+                    if node.item.multiplicity != []:
+                        multiplicity_restricted_templates.append(node)
+            if multiplicity_restricted_templates:
+                for template in multiplicity_restricted_templates:
+                    for struct in product_structures:
+                        match = self._match_reactant_to_template(struct, template)
+                        if match:
+                            break
+                    if not match:
```

Comment:
Complicated nested `for` loops with `if` and `break` clauses etc.  usually benefit greatly from some comments. I think this is correct, but I had to think very hard to check it. Please make life easier for reviewers future programmers by adding some comments here.